### ISSUE TITLE
The geometry does not change on the map when passing from one instance to the other

### ISSUE THREAD

**Issue by SteeveEbener** [terraframe/geoprism-registry#635]
**List the CGR information**
- CGR version: 14.7
- CGR instance (e.g. a URL): Staging

**Describe the bug (clear and concise)**
1. When looking or editing the geometry of a given geo-object presenting several instances we can't pass from one instance to the other in the view/editing window and therefore not see the geometry associated to each instance on the map
2. You can edit the geometry associated to each instance (edit button available for each) but when doing so this is only based on the geometry for the geo-object as it exists on the date for which the list has been extracted
3. The edit done on the geometry for one of the instances is not saved 

**To Reproduce**
As RA/RM for the MOHA in the demo instance:
1. Go to the lists module
2. Open the exploratory lists for villages
3. Click the edit pen for one of the villages
4. Click the edit pen again and the edit button for the geometry (finally there :) )
5. Add a new instance for the geometry (with no temporal overlaps) => example in the first screenshot below
6. First set of issues:
     a. The red dot remains on the map giving the impression the same geometry is saved for both instances. Actually, when you click on the “Edit Geometries” button you see that the red point is attached only to the older instance (no dot appearing for the new instance) while you are actually working on the new instance
     b. The new instance is highlighted in light green (better than the case described here below
     c. You can’t pass from one instance to the other to see how the geometry has changed across the different instances
     d. If you click the “Edit Geometries” button for the new instance, you can’t move the red dot and it is therefore not obvious that you actually have to put a new pin on the map
8. Put a pin on the map, click the save icon on the map => new issue: the pin disappear giving the impression that the location has not been saved in the graph
9. Click the first submit button (point not appearing on the map) then the second one (only now is the new point appearing on the map)

Now, if you apply the same steps as here above but this time you do the following after step 5:
6. Create the new instance, don’t provide the geometry for this new instance and click both submit buttons
7. Put back the geo-object in editing mode and click the edit button for the geometry. 
8. You can now see the two instances (Screenshot No 2 below) and the following first set of issues:
      a. there is no geometry for the new instance (I would have expected the same geometry as for the previous instance to be there as for the above case + it can be that the geometry to be modified is the one for the older instance, not the most recent one) 
      b. The new instance is not highlighted in green like for the first case above 
      c. You still can't pass from one instance to the other to see how the geometry has evolved between the two periods. The only way to see the geometry from the older instance is to click the "Edit Geometries" button
9. Click the Edit geometries button for the new instance
10. Click the pin icon on the map and place the pin anywhere on the map. The orange dot appear on the map
11. Click the save button on the map => same issue as before: the pin disappear giving the impression that the change has not been saved in the graph
12. Click the first submit button (point not appearing on the map) then the second one (only now is the new point appearing on the map)

The same thing happens with geo-object types represented by polygons. For example, have a look at the two instances existing for Andustar Province 

**Expected behavior (clear and concise)**
1.	The experience is the same when the user set the geometry of the new instance before clicking the submit buttons or after having submitted the new instance without setting the geometry (two scenarios described here above after step 5)   
2.	The user can see the geometry of each instance on the map by simply clicking on each of them on the left part of the window. The one being selected should appear highlighted in green like in Screenshot 1 (the other one should remain in white); 
3.	The red dot associated to the instance before the start of the process should only be associated to the older instance

One more thing to consider for the long term: allow the user to decide if he wants to “carry over” the geometry from another instance to the new one (this will be particularly useful for polygon or line type geo-objects)

**Screenshots**
Screenshot 1
![image](https://user-images.githubusercontent.com/12602163/121860730-bb06b980-cd2b-11eb-9da1-0d46231a7304.png)

Screenshot 2
![image](https://user-images.githubusercontent.com/12602163/121860739-be01aa00-cd2b-11eb-8803-ebce0a707bfe.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Google Chrome
 - Version: The latest version

**Additional context (if any)**

**Comment by rrowlands** [comment]
1. The behaviour in the system has changed a bit here as of late. As shown in this screenshot

![image.png](https://images.zenhubusercontent.com/5df81ec613efba2c885c40e6/33ef5ec8-84b3-4b8b-84a3-15a01d7a2a68)

When viewing an object from a list (and the edit button is clicked from the pop-up modal), the system is displaying two different layers:
1. The list in question
2. The geometry of the Geo-Object at the time in question.

If we were to adhoc modify this list layer to not include the geometry of the Geo-Object which was selected from the list, how should this layer behave when:
1. The page is refreshed or a link is given to somebody else, causing that layer to load from the URL parameters and the layer to be loaded.
2. A different Geo-Object is selected from the list layer
3. The geometry of the "list selected Geo-Object" is hidden

Additionally, what should the label of this "adhoc" layer be in the layer panel? Should we append "excluding Geo-Object xxx" to the end of the layer name?

Given that the system is now clearly explaining which layers are being rendered, I feel this point is no longer valid, and I also feel that hiding Geo-Objects will cause more confusion than it will solve.


2. This behaviour is already happening on the latest codebase.

3. Perhaps this discussion would best be had within #674?

4. I believe this is already fixed, yes?

5. This also appears to be already fixed to me.

**Comment by SteeveEbener** [comment]
Hi @rrowlands, 

My answer to the different points:

1 and 2. Please look at my last comment in #736 regarding the expected behavior.  To answer your questions linked to point 1:

- The page is refreshed or a link is given to somebody else, causing that layer to load from the URL parameters and the layer to be loaded - Would implementing #736  address this?
- A different Geo-Object is selected from the list layer - That would mean that you restart from the list, right? I don't see any contradiction with what is reported in #736 
- The geometry of the "list selected Geo-Object" is hidden - What do you mean by hidden? private? 

3. Fine with me

4 and 5. Indeed. Thanks for that

**Comment by justinlewis** [comment]
Closing legacy issues created under a prior funded development effort. These items were associated with a separate roadmap and stakeholder group targeting the v 1.x release. Archiving these tickets enables a v 2 reset of the backlog. All of these tickets can be found by searching for the geoprism-v1-archive tag on tickets. If any issue remains relevant to current priorities, please reopen or ideally create a new issue referencing this thread.
